arviz/plots/pairplot.py

```diff
@@ -142,31 +145,50 @@ def plot_pair(
 
     # Get posterior draws and combine chains
     data = convert_to_inference_data(data)
-    posterior_data = convert_to_dataset(data, group="posterior")
-    var_names = _var_names(var_names, posterior_data)
+    grouped_data = convert_to_dataset(data, group=group)
+    var_names = _var_names(var_names, grouped_data)
     flat_var_names, _posterior = xarray_to_ndarray(
-        get_coords(posterior_data, coords), var_names=var_names, combined=True
+        get_coords(grouped_data, coords), var_names=var_names, combined=True
     )
 
     divergent_data = None
     diverging_mask = None
     # Get diverging draws and combine chains
     if divergences:
-        if hasattr(data, "sample_stats") and hasattr(data.sample_stats, "diverging"):
-            divergent_data = convert_to_dataset(data, group="sample_stats")
-            _, diverging_mask = xarray_to_ndarray(
-                divergent_data, var_names=("diverging",), combined=True
-            )
-            diverging_mask = np.squeeze(diverging_mask)
+        if group == "posterior":
+            if hasattr(data, "sample_stats") and hasattr(data.sample_stats, "diverging"):
+                divergent_data = convert_to_dataset(data, group="sample_stats")
+                _, diverging_mask = xarray_to_ndarray(
+                    divergent_data, var_names=("diverging",), combined=True
+                )
+                diverging_mask = np.squeeze(diverging_mask)
+            else:
+                divergences = False
+                warnings.warn(
+                    "Divergences data not found, plotting without divergences. "
+                    "Make sure the sample method provides divergences data and "
+                    "that it is present in the `diverging` field of `sample_stats` "
+                    "or `sample_stats_prior` or set divergences=False",
+                    SyntaxWarning,
+                )
```

Comment:
I would try to avoid duplicating all this code. How does somethig like this look like? (pseudocode alert)

```
if group=posterior:
    divergent_group = sample_stats
elif group=prior:
    divergent_group = sample_stats_prior
else:
    divergences = False
# Get diverging draws and combine chains
    if divergences:
        if hasattr(data, divergent_group) and hasattr(getattr(data, divergent_group), "diverging"):
...  # divergent_group can also be used to format the warning so it is more informative
```

Reply:
Okay, will do.

Reply:
@OriolAbril I think the code's fine but we should also add some warning if neither `posterior` or `prior` is passed to `group` param when we are setting `divergences=False`. Warning may be something like `Group assigned not in the divergent group --> plotting without divergence`. What do you think about this?